File: Trivial/ProvincesAndGold.java
import java.util.Scanner;

public class ProvincesAndGold {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int g = sc.nextInt();
        int s = sc.nextInt();
        int c = sc.nextInt();
        sc.close();

        int power = 3 * g + 2 * s + c;
        String victory = "";
        String treasure = "Copper";

        if (power >= 8) {
            victory = "Province";
            treasure = "Gold";
        } else if (power >= 5) {
            victory = "Duchy";

            if (power >= 6) {
                treasure = "Gold";
            } else {
                treasure = "Silver";
            }
        } else if (power >= 2) {
            victory = "Estate";

            if (power >= 3) {
                treasure = "Silver";
            } else {
                treasure = "Copper";
            }
        }

        if(!victory.equals("")) {
            System.out.println(victory + " or " + treasure);
        } else {
            System.out.println(treasure);
        }
    }
}

File: solutions/provincesandgold.py
g, s, c = input().split()
g, s, c = int(g), int(s), int(c)

victory_cards = ['Province', 'Duchy', 'Estate']
treasure_cards = ['Gold', 'Silver', 'Copper']

buying_power = 3 * g + 2 * s + c
victory = ''
treasure = treasure_cards[2]

if buying_power >= 8:
    victory = victory_cards[0]
    treasure = treasure_cards[0]
elif buying_power >= 5:
    victory = victory_cards[1]
    if buying_power >=6:
        treasure = treasure_cards[0]
    else:
        treasure = treasure_cards[1]
elif buying_power >= 2:
    victory = victory_cards[2]
    if buying_power >=3:
        treasure = treasure_cards[1]
    else:
        treasure = treasure_cards[2]
    
if buying_power < 2:
    print(treasure)
else:
    print(victory, 'or', treasure)